modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesSubscriptionManager.java

```diff
@@ -322,91 +336,123 @@ public void updateClusterSubscriptionMaps(AndesSubscription subscription, Subscr
+                                    "available in db. Thus removing from memory " + currentSubscription);
+                            subscriptionsToRemove.add(currentSubscription);
+                        }
+                    }
                 }
             }
-
-        }
-        //topics
-        Set<String> topics = subscriptionStore.getAllDestinationsOfSubscriptions(true);
-        for (String topic : topics) {
-            Set<AndesSubscription> memorySubscriptions = subscriptionStore.getAllClusterSubscriptionsWithoutWildcards
-                    (topic, true);
-            Set<AndesSubscription> dbSubscriptions = new HashSet<>();
-            List<String> encodedDBSubscriptions = results.get(TOPIC_PREFIX + topic);
-            if(encodedDBSubscriptions != null) {
-                for (String subscriptionAsStr : encodedDBSubscriptions) {
-                    BasicSubscription subscription = new BasicSubscription(subscriptionAsStr);
-                    dbSubscriptions.add(subscription);
+            //topics
+            Set<String> topics = subscriptionStore.getAllDestinationsOfSubscriptions(true);
+            for (String topic : topics) {
+                Set<AndesSubscription> memorySubscriptions = subscriptionStore.getAllClusterSubscriptionsWithoutWildcards
+                        (topic, true);
+                Set<AndesSubscription> dbSubscriptions = new HashSet<>();
+                List<String> encodedDBSubscriptions = results.get(TOPIC_PREFIX + topic);
+                if(encodedDBSubscriptions != null) {
+                    for (String subscriptionAsStr : encodedDBSubscriptions) {
+                        BasicSubscription subscription = new BasicSubscription(subscriptionAsStr);
+                        dbSubscriptions.add(subscription);
+                    }
                 }
-            }
-            //check if each memory subscription is in DB. If not remove from memory
-            Iterator<AndesSubscription> memSubIterator = memorySubscriptions.iterator();
-            while (memSubIterator.hasNext()) {
-                AndesSubscription currentSubscription = memSubIterator.next();
-                if (!dbSubscriptions.contains(currentSubscription)) {
-                    log.warn("Cluster Subscriptions are not in sync. Subscriptions exist in memory that are not " +
-                            "available in db. Thus removing from memory " + currentSubscription);
-                    subscriptionStore.createDisconnectOrRemoveClusterSubscription(currentSubscription, SubscriptionListener
-                            .SubscriptionChange.DELETED);
-                    notifyClusterSubscriptionHasChanged(currentSubscription, SubscriptionListener.SubscriptionChange.DELETED);
+                //check if each memory subscription is in DB. If not remove from memory
+                if(null != memorySubscriptions) {
```

Comment:
can't we do for (...) loop? ( will bring better clarity ... my personal opinion)